lib/stan_math_2.6.3/stan/math/prim/scal/prob/wiener_log.hpp

```diff
@@ -0,0 +1,224 @@
+    /**
+     * The log of the first passage time density function for a (Wiener) drift diffusion model for the given $y$,
+     * boundary separation $\alpha$, nondecision time $\tau$, relative bias $\beta$, and drift rate $\delta$.
+     * $\alpha$ and $\tau$ must be greater than 0, and $\beta$ must be between 0 and 1. $y$ should contain
+     * reaction times in seconds, with upper-boundary responses strictly positive and lower-boundary response
+     * times coded as strictly negative numbers.
+     *
+     * @param y A scalar variate.
+     * @param alpha The boundary separation.
+     * @param tau The nondecision time.
+     * @param beta The relative bias.
+     * @param delta The drift rate.
+     * @return The log of the Wiener first passage time density of the specified arguments.
+     */
+
+     const double MY_PI = boost::math::constants::pi<double>(),
+     MY_LN_SQRT_PI = std::log(std::sqrt(boost::math::constants::pi<double>())),
+     WIENER_ERR = 0.000001;
+
+     template <bool propto,
+               typename T_y, typename T_alpha, typename T_tau,
+               typename T_beta, typename T_delta>
+     typename return_type<T_y, T_alpha, T_tau, T_beta, T_delta>::type
+     wiener_log(const T_y& y, const T_alpha& alpha, const T_tau& tau,
+       const T_beta& beta, const T_delta& delta) {
+      static const char* function("stan::math::wiener_log(%1%)");
+
+      using stan::math::check_not_nan;
+      using stan::math::check_finite;
+      using stan::math::check_positive;
+      using stan::math::check_bounded;
+      using stan::math::check_consistent_sizes;
+      using stan::math::value_of;
+      using boost::math::tools::promote_args;
+      using boost::math::isinf;
+      using boost::math::isfinite;
+      using std::log;
+      using std::exp;
+      using std::pow;
+
+      if (!(stan::length(y)
+        && stan::length(alpha)
+        && stan::length(beta)
+        && stan::length(tau)
+        && stan::length(delta)))
+        return 0.0;
+
+      typedef typename return_type<T_y, T_alpha, T_tau,
+                                   T_beta, T_delta>::type T_return_type;
+      T_return_type lp(0.0);
+
+      check_not_nan(function, "Random variable", y);
+      check_not_nan(function, "Boundary separation", alpha);
+      check_not_nan(function, "A-priori bias", beta);
+      check_not_nan(function, "Nondecision time", tau);
+      check_not_nan(function, "Drift rate", delta);
+      check_finite(function, "Boundary separation", alpha);
+      check_finite(function, "A-priori bias", beta);
+      check_finite(function, "Nondecision time", tau);
+      check_finite(function, "Drift rate", delta);
+      check_positive(function, "Random variable", y);
+      check_positive(function, "Boundary separation", alpha);
+      check_positive(function, "Nondecision time", tau);
+      check_bounded(function, "A-priori bias", beta , 0, 1);
+      check_consistent_sizes(function,
+       "Random variable", y,
```

Comment:
Need to run int on some of this.  You need at least 4 space indent and ideally just line up under "function".